weasyl/search.py

```diff
@@ -318,22 +342,22 @@ def get_ids(names):
 
     if backid:
         back_count = d.engine.execute(
-            make_statement("SELECT COUNT(*) FROM (SELECT 1", pagination_filter, ") _"), **params).scalar() - len(ret)
+            make_statement("SELECT COUNT(*) FROM (SELECT 1", pagination_filter, " LIMIT %(count_limit)s + %(limit)s) _"), **params).scalar() - len(ret)
```

Comment:
`len(ret)` is subtracted at the end of the statement, because the count includes results that are part of the current page. Adding `limit` here avoids having “10000+ more” on one side and “9937+ more” on the other.